File: solutions/opus-1/opus/javascript/anagrams.js
const data = require('fs').readFileSync(0, 'utf8');
const words = data.split(/\s+/).filter(w => w.length > 0);
const cmp = (a, b) => (a < b ? -1 : a > b ? 1 : 0);
const groups = new Map();
for (const w of words) {
  const key = w.split('').sort().join('');
  if (!groups.has(key)) groups.set(key, []);
  groups.get(key).push(w);
}
const out = [...groups.values()].map(g => g.sort(cmp));
out.sort((a, b) => cmp(a[0], b[0]));
process.stdout.write(out.map(g => g.join(' ') + '\n').join(''));

File: solutions/opus-1/opus/perl/anagrams.pl
use strict;
use warnings;

my %groups;
while (my $line = <STDIN>) {
    $line =~ s/\s+//g;
    next if $line eq '';
    my $key = join '', sort split //, $line;
    push @{ $groups{$key} }, $line;
}

my @out = map { [ sort @$_ ] } values %groups;
for my $g (sort { $a->[0] cmp $b->[0] } @out) {
    print join(' ', @$g), "\n";
}
